TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -86,13 +86,21 @@ public static IEnumerable<CompletionItem> CreateCompletionItemsForType(IEnumerab
                 var typeDisplayName = typeIsPublic ? type.VisualQualifiedName.ToString() : type.Name;
                 var completionItem = new CompletionItem(typeDisplayName);
 
-                completionItem.insertText = typeIsPublic
-                    ? node is FunctionDeclaration
-                        ? string.Format("{0}::{1}", type.VisualQualifiedName.Tail, type.VisualQualifiedName.Head)
-                        : string.Format("{0}::{1}.", type.VisualQualifiedName.Tail, type.VisualQualifiedName.Head)
-                    : node is FunctionDeclaration
-                        ? type.Name 
-                        : type.Name + ".";
+                if (!(node is FunctionDeclaration))
+                    if (typeIsPublic)
+                    {
+                        completionItem.insertText =
+                            node.CodeElement.ConsumedTokens.Last().SourceText == "." //Check if last element is a point, so the completion does not make a duplicate
```

Comment:
You should better use the type of the ConsumedToken. A "." is of type PeriodSeparator